File: c/two-sum/two-sum.c
/**
 方法1 暴力遍历法
 */
int* twoSum(int* nums, int numsSize, int target, int* returnSize){
    int i = 0;
    int j = 0;
    *returnSize = 2;
    int * ret = (int *)malloc(sizeof(int) * 2);
    for(i = 0; i < numsSize - 1; i++){
        for(j = i + 1; j < numsSize; j++){
            if(target - nums[i] == nums[j]) {
                ret[0] = i;
                ret[1] = j;
            }
        }
    }    
    return ret;
}

File: golang/two-sum/two-sum.go
/*
hash查找法
*/

func twoSum(nums []int, target int) []int {
    nummap := make(map[int]int)

    for i, num := range nums {
        nummap[num] = i
    }
    for i, num := range nums {
        index := target - num
        if _, exist := nummap[index]; exist && nummap[index] != i{
            return []int{i, nummap[index]}
        }
    }
    return []int{}
}